hal/include/HAL/DIO.h

```diff
@@ -12,29 +12,27 @@
 #include "Handles.h"
 
 extern "C" {
-// the following 2 functions are here as they will be changed with
-// the handle changes to be DIO exclusive.
-void* initializeDigitalPort(HalPortHandle port_handle, int32_t* status);
-void freeDigitalPort(void* digital_port_pointer);
+HalDigitalHandle initializeDIOPort(HalPortHandle port_handle, uint8_t input,
```

Comment:
We were talking about not wanting bool in the C HAL API. We were going to wait until the rename to change that everywhere else. But I figured a new variable would be fine with that.